pkg/rtc/transportmanager.go

```diff
@@ -607,7 +607,7 @@ func (t *TransportManager) handleConnectionFailed(isShortLived bool) {
 
 	lastSignalSince := time.Since(t.lastSignalAt)
 	signalValid := t.signalSourceValid.Load()
-	if lastSignalSince > iceFailedTimeout || !signalValid {
+	if !t.hasRecentSignalLocked() || !signalValid {
```

Comment:
this was the problem, in the repro that I had, lastSingalSince was 5.1s, slightly higher than iceFailedTimeout. I've moved this to be aligned with ping timeout.

Reply:
we add this because we don't want to switch candidate type when it is caused by the network interruption(both signal and media transport were failed), if the ping timeout (20s) greater than icetotaltimeout(15s), it might not work well in this case.

Reply:
I see. What would be a better way to coordinate this? It seems that signal liveliness shouldn't be using ice timeout values? Other than ping, it isn't necessary to have any signal traffic after the offer/answer

Reply:
I think we can use a shorter hearbeat 5s and then use icetotaltimeout to check if signal is interrupted at same time with ice

Reply:
changed to shorter heartbeat, and aligned signal timeout to the same values as total ICE timeout